scripts/source-maintenance.sh

```diff
@@ -40,6 +40,26 @@ fi
 COPYRIGHT_YEARS=`date +"1996-%Y"`
 echo "s/1996-2[0-9]+ The Squid Software Foundation and contributors/${COPYRIGHT_YEARS} The Squid Software Foundation and contributors/g" >>boilerplate_fix.sed
 
+updateifchanged()
+{
+original="$1"
+updated="$2"
+message="$3"
+
+if test -e "${original}" -a -e "${updated}"; then
+	md51=`cat "${original}" | tr -d "\n \t\r" | ${MD5}`;
+	md52=`cat "${updated}" | tr -d "\n \t\r" | ${MD5}`;
+	if test "${md51}" != "${md52}" ; then
+		echo "NOTICE: File ${original} changed: ${message}"
```

Comment:
IIRC it was because whitespace only changes were quite common when these scripts were added. I am retaining it here for consistency with the existing notices.

Reply:
Thank you for explaining your rationale. Armed with that understanding, I can advance this change request further: Please replace this detection of non-whitespace changes with detection of any changes (using something like `diff -q $original $updated`). Please see below for justification.

The actual reasons why the existing/official code removes whitespace are very different from "whitespace only changes were quite common [and should be ignored]":

1. Commit 1afc2b9 introduced that `tr|md5` trick in order to detect astyle bugs. You even mentioned the trick in that commit message. Astyle was not supposed to modify anything but the whitespace so our trick worked well. Only whitespace changes are (silently) allowed. Non-whitespace changes are undone (with a warning).
2. Commit 987ca90 reused the trick to warn about poorly ordered `#include`s while not warning about (but still applying!) whitespace-only changes. All changes are applied. Non-whitespace changes generate a warning.

In both cases, applying the proposed PR logic would result in bugs -- each of those special existing cases needs a _different_ treatment. As for future plugins, some of them will likely need a different default treatment as well. In summary, the current PR default cannot be justified by the old code and is unlikely to be correct for at least some future code. It should be changed.

I hope the above detailed explanation clarifies why a simpler "any changes count" default should be used instead. Ideally, the plugin should signal its high-level checking needs somehow, but this change request is _not_ about that advanced feature.

Reply:
Okay. Done.